File: Baekjoon/수학/9506.py
# https://www.acmicpc.net/problem/9506

import sys
input = sys.stdin.readline

while True:
    n = int(input())
    if n == -1:
        break
    divisors = []
    for i in range(1, n // 2 + 1):
        if n % i == 0:
            divisors.append(i)
    if sum(divisors) == n:
        print(f"{n} = {' + '.join(map(str, divisors))}")
    else:
        print(f"{n} is NOT perfect.")

File: Baekjoon/수학/9506.cpp
// https://www.acmicpc.net/problem/9506

#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    while (true) {
        int n;
        cin >> n;
        if (n == -1) break;
        vector<int> divisors;
        int sum = 0;
        for (int i = 1; i < n; i++) {
            if (n % i == 0) {
                divisors.push_back(i);
                sum += i;
            }
        }
        if (sum == n) {
            cout << n << " = ";
            for (int i = 0; i < divisors.size(); i++) {
                cout << divisors[i];
                if (i != divisors.size() - 1) cout << " + ";
            }
            cout << '\n';
        } else {
            cout << n << " is NOT perfect.\n";
        }
    }
    return 0;
}